lib/json_api_controller/relation_manager.rb

```diff
@@ -31,16 +31,13 @@ def destroy_relation(resource, relation, value)
 
     def find_for_string_type(resource, type, id)
       relation_find(id) do
-        type.camelize
-          .singularize
-          .constantize
-          .link_to_resource(resource, api_user)
+        Pundit.policy!(api_user, resource).linkable_for(type)
```

Comment:
It doesn't generate a `PolicyScope`, generates the Policy itself. It's basically `ProjectPolicy.new(api_user, resource).linkable_workflows`. It's an instance method on the policy so that it can pass in some filter based on the resource to the `WorkflowPolicy::Scope`, e.g. scoping it to only workflows belonging to the project.

Where would your `linkable_model_resources_for_type` exist as a method?

The main thing I would see as confusing myself is that `ProjectPolicy.new`, and thus by extension `Pundit.policy!`, can take both model instances (what would be returned by `Project.find(1)`) as well as AR relation objects (i.e. `Project.where(active: true)`).

Reply:
yeah - I used the wrong wording in my comment, i get that a Policy object runs the linkable_methods. I find the wording of `Pundit.policy!(api_user, resource).linkable_for(type)` hard to understand what it's doing, maybe splitting the policy creation helps. `linkable_for` doesn't really help me here either, `policy.linkable_for(type)`
> Where would your linkable_model_resources_for_type exist as a method?

In the `relation_manager.rb`for re-use and to capture the intention of the resolving linking AR scopes i have access to on the linking resource / type. As i said above interpreting the pundit line directly is hard.

Reply:
In theory because both RelationManager and PunditPolicy are modules mixed into the controller, in here we could just do `policy.linkable_for`.

Suggestions (either both or just one):

1. Convert RelationManager into a service object instead of a mixin, passing in the resource and the policy object.
2. Rename `linkable_for`, perhaps `association_scope`?

Reply:
anything to relay the intention away from how it is now is ideal. option 2 makes this clearer to me. Understanding how the linkable scopes are wired up via these methods is key to ensuring clarity long term. 

Also knowing which way the relations are wired up will help, e.g. resource vs type. i'd probably have to run the code to know which way these variables are actually passed in. if you can rename these to be clearer `resource_to_link_to` and `type_to_link` could help (if i have that the right way around).